scipy/stats/_multivariate.py

```diff
@@ -2648,6 +2661,394 @@ def entropy(self):
+}
+
+class multinomial_gen(multi_rv_generic):
+    r"""
+    A multinomial random variable.
+
+    .. versionadded:: 0.18.0
+
+    Methods
+    -------
+    ``pmf(x, n, p)``
+        Probability mass function.
+    ``logpmf(x, n, p)``
+        Log of the probability mass function.
+    ``rvs(n, p, size=1, random_state=None)``
+        Draw random samples from a multinomial distribution.
+    ``entropy(n, p)``
+        Compute the entropy of the multinomial distribution.
+    ``cov(n, p)``
+        Compute the covariance matrix of the multinomial distribution.
+
+    Parameters
+    ----------
+    x : array_like
+        Quantiles, with the last axis of `x` denoting the components.
+    %(_doc_default_callparams)s
+    %(_doc_random_state)s
+
+    Notes
+    -----
+    %(_doc_callparams_note)s
+
+    Alternatively, the object may be called (as a function) to fix the `n` and
+    `p` parameters, returning a "frozen" multinomial random variable:
+
+    The probability mass function for `multinomial` is
+
+    .. math::
+
+        f(x) = \begin{pmatrix} n \\ x_1, \ldots, x_k \end{pmatrix}
+        p_1^{x_1} \cdots p_k^{x_k},
+
+    supported on :math:`x=(x_1, \ldots, x_k)` where each :math:`x_i` is a
+    nonnegative integer and their sum is :math:`n`.
+
+    Examples
+    --------
+
+    >>> from scipy.stats import multinomial
+    >>> rv = multinomial(8, [0.3, 0.2, 0.5])
+    >>> rv.pmf([1, 3, 4])
+    0.042000000000000072
+
+    The functions `pmf`, `logpmf`, `entropy`, and `cov` support broadcasting,
+    under the convention that the vector parameters (`x` and `p`) are
+    interpreted as if each row along the last axis is a single object. For
+    instance:
+
+    >>> multinomial.pmf([[3, 4], [3, 5]], [7, 8], [.3, .7])
+    array([0.2268945,  0.25412184])
+
+    Here, `x.shape == (2, 2)`, `n.shape == (2,)`, and `p.shape == (2,)`, but
+    following the rules mentioned above they follow numpy's broadcasting rules
+    as if the rows `[3, 4]` and `[3, 5]` in `x` and `[.3, .7]` in `p` were a
+    single object, and as if we had `x.shape = (2,)`, `n.shape = (2,)`, and
+    `p.shape = (,)`.
+
+    This broadcasting also works for `cov`, where the output objects
+    are square matrices of size `p.shape[-1]`. For example:
+
+    >>> multinomial.cov([4, 5], [[.3, .7], [.4, .6]])
+    array([[[ 0.84, -0.84],
+            [-0.84,  0.84]],
+           [[ 1.2 , -1.2 ],
+            [-1.2 ,  1.2 ]]])
+
+    In this example, `n.shape == (2,)` and `p.shape == (2, 2)`, and
+    following the rules above, these broadcast as if
+    `p.shape == (2,)`. Thus the result should also be of shape
+    `(2,), but since each output is a :math:`2 \times 2` matrix, the result
+    in fact has shape `(2, 2, 2)`.
+
+    Currently, `rvs` does not support broadcasting. Instead, the first element
+    of `n` and the first vector in `p` are used to choose the distribution from
+    which to sample. The current behavior should not be relied on.
```

Comment:
Yuk. Please, no. If it's not supported, it should raise an error with an informative message. Better still, it could probably loop over the input, but that can be left as an optional future enhancement.